api/src/opentrons/protocol_runner/json_translator.py

```diff
@@ -106,31 +132,40 @@ def _translate_simple_command(
     return translated_obj
 
 
-class JsonCommandTranslator:
-    """Class that translates commands from PD/JSON to ProtocolEngine."""
-
-    def translate(
+class JsonTranslator:
+    """Class that translates commands/liquids from PD/JSON to ProtocolEngine."""
+
+    def translate_liquids(self, protocol: ProtocolSchemaV6) -> List[Liquid]:
+        """Takes json protocol v6 and translates liquids->protocol engine liquids."""
+        liquids = []
+        if protocol.liquids:
+            for liquid_key in protocol.liquids:
+                liquid = protocol.liquids[liquid_key]
+                liquids.append(
+                    Liquid(
+                        id=liquid_key,
+                        displayName=liquid.displayName,
+                        description=liquid.description,
+                        displayColor=liquid.displayColor,
+                    )
+                )
+        return liquids
```

Comment:
Prefer for comprehensions

```suggestion
        protocol_liquids = protocol.liquids or {}
        
        return [
            Liquid(
                 id=liquid_id,
                 displayName=liquid.displayName,
                 description=liquid.description,
                 displayColor=liquid.displayColor,
            )
            for liquid_id, liquid in protocol_liquids.items()
        ]
```